source/simulator/helper_functions.cc

```diff
@@ -1839,6 +1839,7 @@ namespace aspect
     // updating the ghost elements of the 'solution' vector.
     const unsigned int advection_block = adv_field.block_index(introspection);
     distributed_vector.block(advection_block).compress(VectorOperation::insert);
+    current_constraints.distribute (solution);
```

Comment:
That doesn't seem right. Either apply to distributed_vector or do this after the last line.

Reply:
Thank you, Timo! I corrected ir in the latest commit.